platform/default/mbgl/storage/offline_download.hpp

```diff
@@ -49,7 +49,8 @@ class OfflineDownload {
      */
     void ensureResource(const Resource&, std::function<void (Response)> = {});
     void ensureTiles(SourceType, uint16_t, const SourceInfo&);
-
+    bool hasExceededTileCount(const Resource& resource);
```

Comment:
This method has side effects, but `hasExceededTileCount` sounds like the name of a pure function. How about `checkTileCountLimit`?